state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1854,6 +1869,48 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Ok(action)
 	}
 
+	/// Checks if the transaction is tainted and schedules it for rejection if it is.
+	fn check_if_tx_is_tainted_and_schedule_reject(
+		channel_owner: T::AccountId,
+		deposit_details: <T::TargetChain as Chain>::DepositDetails,
+		deposit_amount: TargetChainAmount<T, I>,
+		asset: TargetChainAsset<T, I>,
+		deposit_address: TargetChainAccount<T, I>,
+		action: ChannelAction<T::AccountId>,
+	) -> bool {
+		if TaintedTransactions::<T, I>::take(channel_owner.clone(), deposit_details.clone())
```

Comment:
Nit, but it seems like we are doing way to many clones here. For example, when you call `check_if_tx_is_tainted_and_schedule_reject` you preemptively clone `deposit_address` but in the common case of "transaction not tainted" that value isn't even used. Also, I believe you can pass references to `take` in this line. I haven't tested this, but conceptually, we shouldn't clone anything (including at check_if_tx_is_tainted_and_schedule_reject`'s callers side) unless the transaction is in fact tainted and you need to create new objects like `TaintedTransactionDetails` and `DepositIgnored`.